### DISCUSSION TITLE
OpenToonz Blender Bridge

### DISCUSSION THREAD

**Discussion by adamearle** [opentoonz/opentoonz#6510] [Ideas]
Working with 2D and 3D has been proven through history to help create more cinematic animation. OT Bridge would bring starting with Blender bring that capability to the opensource world. Combing the capability of both softwares will only enable artist to come up with a more accessible and cinematic workflow to be explored and build upon. 

Below are some link of how Toonz now OpenToonz will was used in the pass
1. [https://www.youtube.com/watch?v=kj8ndIraGp0](https://www.youtube.com/watch?v=kj8ndIraGp0)
2. https://www.youtube.com/watch?v=Dze5SYKtLsc

### Proposed Starting idea:

** Send to "X (Blender)" ** 

1. "Send to "X (Blender)" Button
2. Folder added to the Opentoonz project structure
3. Blender import the level files from the OpenToonz into Blender (or render PNGs (Quality settings should be set here from roughing to final)
4. Import images as plane add-on code can be used, the correct material set and the plane attached to the camera or placed in the right position in space. 
5. The object plane that is attached to the camera or position in space is named the same as the level Column in Opentoonz
6. The position of the camera is transferred across.
7. (Vectors could eventually could be converted to grease pencils).

** Send to OpenToonz**

1. If the object plane is renamed in Blender on update the a column popup window 
2. Will auto-fill to the current project 
3. Or give you the opportunity to create a new project level
4. Ask you to select what level you are updating too
5. Or create a new level
6. Change and any of the position modification made in space and send them across to OpenToonz

**Comment by adamearle** [comment]
@shun-iwasawa same thing here as well I am looking for outside help. 

What are your thoughts on this?

Is there anything in this that you think you can tackle?

Cheers Ad

**Comment by adamearle** [comment]
Also contacted the director from Tekkonkinkreet for his thoughts on this as well. There may be a better way fo doing things now? Tracking and point clouds?

**Comment by artisteacher** [comment]
It seems to me that a lot of this is really Blender specific & would need to be done by people familiar with the Blender codebase.

**Comment by ave4** [comment]
For this I think the RenderChan https://github.com/morevnaproject-org/RenderChan program, which develops @morevnaproject, came up. It is intended primarily for the automation of rendering, including the gluing of different software within one project.
As far as I know morevnaproject is used to combine blender 3d animation and 2d animation by Synfig Studio and Krita.
I plan to add Opentoonz support to RenderChan.
https://github.com/morevnaproject-org/RenderChan/issues/27

**Comment by morevnaproject** [comment]
On that topic - http://www.toonz.com/htm/products/46productline/prod3dp.htm

> The plug-in reads the movements of the camera, objects and changes of the camera "field of view" in the 3D environment; then it computes the projections in the 2D Toonz environment.

**Comment by tonypuglieso** [comment]
Hi I just wanted to port that there is a plugin for krita that bridge makes with blender, it can be a good gateway to understand how to make it in opentoonz. https://github.com/Yuntokon/BlenderLayer
